tests/p2p/test_connections.py

```diff
@@ -26,15 +24,14 @@ def test_connections(self):
     def test_manager_connections(self):
         tx_storage = TransactionMemoryStorage()
         tmpdir = tempfile.mkdtemp()
+        self.tmpdirs.append(tmpdir)
         wallet = Wallet(directory=tmpdir)
         wallet.unlock(b'teste')
         manager = HathorManager(self.clock, tx_storage=tx_storage, wallet=wallet)
 
         endpoint = 'tcp://127.0.0.1:8005'
         manager.connections.connect_to(endpoint, use_ssl=True)
 
-        self.assertNotIn(endpoint, manager.connections.iter_not_ready_endpoints())
-        self.assertNotIn(endpoint, manager.connections.iter_ready_connections())
-        self.assertNotIn(endpoint, manager.connections.iter_all_connections())
-
-        shutil.rmtree(tmpdir)
```

Comment:
No need to remove anymore?

Reply:
We _do_ need to remove it, however doing it on the test itself is bad because if the test fails it will not reach that line. I was going to port it to setUp/tearDown which is where these things should be, but I noticed that we already have a mechanism to do that, anything that is added to `self.tmpdirs` will be removed on `tearDown`, so I did just that.